skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+        
+    def _partial_fit(self, X, y, weight):
+        self.instances_seen += 1
+        
+        if self.ensemble is None:
+            self.init_ensemble(X)
+
+        for i in range(self.nb_ensemble):
+            y_predicted = self.ensemble[i].predict(np.asarray([X]))
+            self.ensemble[i].evaluator.update(y_predicted, np.asarray([y]), weight)
+            k = np.random.poisson(self.lambda_value)
+            if k > 0:
+                self.ensemble[i].partial_fit(np.asarray([X]), np.asarray([y]), np.asarray([k]), self.instances_seen)
+    
+    def predict(self, X):
+        """Predicts the label of the X instance(s)
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (n_samples, n_features)
+            Samples for which we want to predict the labels.
+        Returns
+        -------
+        list
+            Predicted labels for all instances in X.
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i])
+            if votes == {}:
+                # Ensemble is empty, all classes equal, default to zero
+                predictions.append(0)
+            else:
+                # predict = []
+                # for vote in votes:
+                predictions.append(max(votes, key=votes.get))  # TODO Verify approach
+                # y, counts = np.unique(predict, return_counts=True)
+                # value = np.argmax(counts)
+                # predictions.append(y[value])
+        return predictions
+
+    def predict_proba(self, X):
+        raise NotImplementedError
+        
+    def reset(self):        
+        """Reset ARF."""
+        self.ensemble = None
+        self.subspace_size = 0
+        self.instances_seen = 0
+        self._train_weight_seen_by_model = 0.0
+        self._evaluator_method = ARFBaseClassifierEvaluator
+        
+    def score(self, X, y):
+        raise NotImplementedError
+        
+    def get_info(self):
+        return "NotImplementedError"
+        
+    def get_votes_for_instance(self, X):
+        if self.ensemble is None:
+            self.init_ensemble(X)
+        combined_votes = {}
+
+        for i in range(self.nb_ensemble):
+            vote = self.ensemble[i].get_votes_for_instance(X)
+            if vote != {} and sum(vote.values()) > 0:    # TODO check vote normalization
+                normalize_values_in_dict(vote)
+                for k in vote:
+                    try:
+                        combined_votes[k] += vote[k]
+                    except KeyError:
+                        combined_votes[k] = vote[k]
+        return combined_votes
+        
+    def init_ensemble(self, X):
+        self.ensemble = [None] * self.nb_ensemble
+
+        # The m (total number of attributes) depends on:
+        _, n = get_dimensions(X)
+        
+        if self.feature_mode == FEATURE_MODE_SQRT:
```

Comment:
This part closely resembles the java version, but probably it could be simplified by using the same approach as in RandomForestClassifier

Reply:
Agree.
- [x] Moved code into own function _set_max_features, providing same options as scikit-learn RandomForestClassifier